lib/byron/src/Cardano/Wallet/Byron/Network.hs

```diff
@@ -680,6 +744,96 @@ chainSyncWithBlocks tr bp queue channel = do
+-- of these clients is necessary, rather than one client per wallet.
+chainSyncFollowTip
+    :: forall m protocol.
+        ( protocol ~ ChainSync ByronBlock (Tip ByronBlock)
+        , MonadThrow m, MonadST m
+        )
+    => Tracer m (TraceSendRecv protocol)
+        -- ^ Base tracer for the mini-protocols
+    -> W.BlockchainParameters
+        -- ^ Blockchain parameters
+    -> (Tip ByronBlock -> m ())
+        -- ^ Callback for when the tip changes.
+    -> Channel m ByteString
+        -- ^ A 'Channel' is a abstract communication instrument which
+        -- transports serialized messages between peers (e.g. a unix
+        -- socket).
+    -> m Void
+chainSyncFollowTip tr bp onTipUpdate channel =
+    runPeer tr codec channel (chainSyncClientPeer client)
+  where
+    -- fixme: it's not necessary to deserialise blocks here, so
+    -- Ouroboros.Consensus.Network.NodeToClient.defaultCodecs could be used.
+    codec :: Codec protocol DeserialiseFailure m ByteString
+    codec = chainSyncCodec bp
+
+    client :: ChainSyncClient ByronBlock (Tip ByronBlock) m Void
+    client = ChainSyncClient (clientStIdle False)
+      where
+        -- Client in the state 'Idle'. We immediately request the next block.
+        clientStIdle
+            :: Bool
+            -> m (ClientStIdle ByronBlock (Tip ByronBlock) m Void)
+        clientStIdle synced = pure $ SendMsgRequestNext
+            (clientStNext synced)
+            (pure $ clientStNext synced)
+
+        -- In the CanAwait state, we take the tip point given by the node and
+        -- ask for the intersection of that point. This fast-fowards us to the
+        -- tip. Once synchronised with the tip, we expect to be waiting for the
+        -- server to send AwaitReply most of the time.
+        clientStNext
+            :: Bool
+            -> ClientStNext ByronBlock (Tip ByronBlock) m Void
+        clientStNext False = ClientStNext
+                { recvMsgRollBackward = const findIntersect
+                , recvMsgRollForward = const findIntersect
+                }
+          where
+            findIntersect tip = ChainSyncClient $
+                pure $ SendMsgFindIntersect [getTipPoint tip] clientStIntersect
+
+        clientStNext True = ClientStNext
+                { recvMsgRollBackward = const doUpdate
+                , recvMsgRollForward = const doUpdate
+                }
+          where
+            doUpdate tip = ChainSyncClient $ do
+                onTipUpdate tip
+                clientStIdle True
+
+        -- After an intersection is found, we return to idle with the sync flag
+        -- set.
+        clientStIntersect
+            :: ClientStIntersect ByronBlock (Tip ByronBlock) m Void
+        clientStIntersect = ClientStIntersect
+            { recvMsgIntersectFound = \_intersection _tip ->
+                ChainSyncClient $ clientStIdle True
+            , recvMsgIntersectNotFound = \_tip ->
+                ChainSyncClient $ clientStIdle False
+            }
```

Comment:
Okay, I think this works fine because the node will always start by replying with a `Rollbackward` to a `RequestNext`. Without that, we would have to wait at least for the next request to override the initial value for the tip. 

:+1:

Reply:
Yes the both the `RollForward` and `RollBackward` messages gives us a new tip, so we update the TVar and then ask the node what's next.